server/paymentProviders/stripe/creditcard.ts

```diff
@@ -82,15 +82,16 @@ const createChargeAndTransactions = async (
     });
   }
 
-  paymentIntent = await stripe.paymentIntents.confirm(paymentIntent.id, {
-    stripeAccount: hostStripeAccount.username,
-  });
+  paymentIntent = await stripe.paymentIntents.confirm(
+    paymentIntent.id,
+    { payment_method: stripePaymentMethod.id },
+    { stripeAccount: hostStripeAccount.username },
+  );
 
   /* eslint-enable camelcase */
 
   if (paymentIntent.next_action) {
-    order.data.paymentIntent = { id: paymentIntent.id, status: paymentIntent.status };
-    await order.update({ data: order.data });
+    await order.update({ data: { ...order.data, paymentIntent } }); // Store the payment intent to make sure it will be re-used after the 3D secure confirmation
```

Comment:
Here the order was not updated successfully. I believe it's an issue with sequelize doing a shallow comparison (rather than a deep comparison) on `data`; thus trying to update without copying the object doesn't have any effect.

A small snippet if you want to verify that assertion for yourself:

```es6
setTimeout(() => {
  models.Order.findByPk(6442).then(o => {
    o.data.test = 1;
    console.log('(1) Has changes?', o.changed());

    o.data = { ...o.data, test: 2 };
    console.log('(2) Has changes?', o.changed());
  });
}, 3000);
```

Result:

> (1) Has changes? false
> (2) Has changes? [ 'data' ]